libs/api/domains/auth/src/lib/services/consentTenants.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@nestjs/common'
+
+import { Auth, AuthMiddleware, User } from '@island.is/auth-nest-tools'
+import { ScopeNodeDTO, ScopesApi } from '@island.is/clients/auth/delegation-api'
+
+import { ConsentScopeNode } from '../models/consentScopeNode.model'
+import { ConsentTenant } from '../models/consentTenants.model'
+
+@Injectable()
+export class ConsentTenantsService {
+  constructor(private readonly scopesApi: ScopesApi) {}
+
+  private scopesApiWithAuth(auth: Auth) {
+    return this.scopesApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  private scopeNodeMapper(
+    node: ScopeNodeDTO,
+    consentedScopes: string[],
+  ): ConsentScopeNode {
+    return {
+      name: node.name,
+      displayName: node.displayName,
+      description: node.description,
+      children: node.children
+        ? node.children.map((child) =>
+            this.scopeNodeMapper(child, consentedScopes),
+          )
+        : [],
+      hasConsent: consentedScopes.includes(node.name),
+    }
+  }
+
+  async getPermissions(
```

Comment:
This should be able to be similar to the implementation in the [apiScope.service](https://github.com/island-is/island.is/blob/main/libs/api/domains/auth/src/lib/services/apiScope.service.ts#L31)